expanding-list-web-component/main.js

```diff
@@ -4,60 +4,57 @@ class ExpandingList extends HTMLUListElement {
     // Always call super first in constructor
     // Return value from super() is a reference to this element
     self = super();
+  }
 
+  connectedCallback() {
     // Get ul and li elements that are a child of this custom ul element
     // li elements can be containers if they have uls within them
-    const uls = Array.from(self.querySelectorAll('ul'));
-    const lis = Array.from(self.querySelectorAll('li'));
-
+    const uls = Array.from(self.querySelectorAll("ul"));
+    const lis = Array.from(self.querySelectorAll("li"));
     // Hide all child uls
     // These lists will be shown when the user clicks a higher level container
-    uls.forEach(ul => {
-      ul.style.display = 'none';
+    uls.forEach((ul) => {
+      ul.style.display = "none";
     });
 
     // Look through each li element in the ul
-    lis.forEach(li => {
+    lis.forEach((li) => {
       // If this li has a ul as a child, decorate it and add a click handler
-      if (li.querySelectorAll('ul').length > 0) {
+      if (li.querySelectorAll("ul").length > 0) {
         // Add an attribute which can be used  by the style
         // to show an open or closed icon
-        li.setAttribute('class', 'closed');
+        li.setAttribute("class", "closed");
 
         // Wrap the li element's text in a new span element
         // so we can assign style and event handlers to the span
         const childText = li.childNodes[0];
-        const newSpan = document.createElement('span');
+        const newSpan = document.createElement("span");
 
         // Copy text from li to span, set cursor style
         newSpan.textContent = childText.textContent;
-        newSpan.style.cursor = 'pointer';
-        
+        newSpan.style.cursor = "pointer";
+
         // Add click handler to this span
-        newSpan.onclick = self.showul;
-        
+        newSpan.addEventListener("click", (e) => {
+          // next sibling to the span should be the ul
+          const nextul = e.target.nextElementSibling;
```

Comment:
camelCase would be more consistent `nextUL`

Reply:
I see your feedback @OnkarRuikar :) - however I'm going to merge just as content & examples are currently out of sync - possibly worth raising a further issue 👍